src/imageLoader/wadors/metaDataManager.js

```diff
@@ -11,7 +71,26 @@ function add(imageId, metadata) {
 function get(imageId) {
   const imageURI = imageIdToURI(imageId);
 
-  return metadataByImageURI[imageURI];
+  let metadata = metadataByImageURI[imageURI];
+
+  if (!metadata) {
+    // in this case try see if the imageId corresponds to multiframe
+    const imageIdFrameless = imageURI.slice(
+      0,
+      imageURI.indexOf('/frames/') + 8
+    );
+    const frame = parseInt(
+      imageURI.slice(imageURI.indexOf('/frames/') + 9),
+      10
+    );
+
+    metadata = metadataByImageURI[`${imageIdFrameless}1`];
+    if (metadata) {
+      metadata = combineFrameInstance(frame, metadata);
+    }
```

Comment:
Actually we shouldn't do this operation on `get`, it should be done on `add` and `get` should be just returning the correct key. If you do it on `get`, all the combining effort is repeated every time.

Reply:
The dicomweb client will return only one imageid/metadata for a multiframe serie. So the user at first will have only one metadata to add to cswil. That why I choose to add one metadata and return n frames in get

Reply:
What is the imageId that is getting returned from dicomweb client? Can you give me an example?

Reply:
The dicomweb client returns just one metadata information and I generate it as this formula
imageId =
      prefix +
      wadoRsRoot +
      '/studies/' +
      StudyInstanceUID +
      '/series/' +
      SeriesInstanceUID +
      '/instances/' +
      SOPInstanceUID +
      '/frames/1';